apps/site/lib/site_web/controllers/stop_controller.ex

```diff
@@ -480,4 +480,18 @@ defmodule SiteWeb.StopController do
 
     %{conn | assigns: assigns}
   end
+
+  def get_facilities(conn, %{"stop_id" => stop_id}) do
+    res =
+      [{"stop", stop_id}]
+      |> V3Api.Facilities.filter_by()
+
+    case res do
+      {:error, _x} ->
+        json(conn, res)
```

Comment:
**suggestion:** Elixir Tuples are not JSON-serializable, so this will break. And are you sure you want to send the error to the frontend? Could return a reasonable fallback value instead.

Reply:
The pattern I've noticed is sending the response whether it's an error or not and handling in the frontend

Reply:
I added an [encoder](https://github.com/mbta/dotcom/blob/master/apps/site/lib/site_web/encoding/tuple_encoder.ex) a few months back that should handle tuples.  That being said I think the ideal case (for production) would be just to return a 500 status code.  Then in dev it could return a status code and an error describing what happened